AEXS/aex-4.md

```diff
@@ -0,0 +1,66 @@
+# AEX-4
+
+```
+AEX: 4
+Title: æternity wallet deep linking specification
+Author: Shubhendu Shekhar (@shekhar-shubhendu), Andrea Giacobino (@noandrea)
+License: BSD-3-Clause
+Status: Draft
+Type: Standards Track
+Created: 2019-04-03
+```
+
+## Simple Summary
+
+The document describes and defines the deep linking specification that every æternity supported wallet can implement and follow to redirect users to a specific activity or page inside the application.
+
+## Motivation
+
+URI based deep linking enables websites or applications to interact with native applications registered to listen for aeternity URI scheme and trigger wallet actions like open wallet accounts directly from third-party applications.
+
+This also enables users to sign transaction using desktop or app based wallets without the need to copy or remember long addresses.
+This is especially beneficial for mobile wallet users as they can simply click on the wallet deeplink to open the wallet app from the vendor app, confirm the transaction and return back to the vendor app through the callback url.
+
+## Specification
+
+### URI Scheme
+
+- `ae:`
+- `aeternity:`
+
+### Methods
+
+1. View/Open Account in the wallet
+
+    URI: `aeternity:<wallet_address>`
+
+    Associated wallet app will:
+
+    1. Open the wallet app
+    2. Check if the address exists or not
+    3. If the acoount is found with the wallet, then the user is redirected to the account information page of the provided address.
+    4.If the account is not found then prompt the user with an `account not found` message.
+
+    #### Example
+
+    `aeternity:ak_2i2fioFMoEffBPeT3EBZEsxK1w579BuCgYE8WiMiADEQqUguU2`
+
+2. Payment
+
+    URI: `aeternity:<receiver_address>/<amount>/<vendor_name>?callback=<url>&txId=<transaction_id>`
+
+    This URI scheme enables vendors to generate direct payment request URIs to user wallet.
+
+    1. Vendor generates a address for receiving payment.
+    2. Creates the URI substituting `receiver_address` with newly created address, `amount` with amount requested, `vendor_name` with unique self-identifier for the user to correctly identifiy vendor and `url` with a callback URL that accepts a transaction id under `txId` query param.
```

Comment:
Where does this `unique self-identifier` come from? And in what context is it unique?

Reply:
`unique self-identifier` is provided by the vendor. it is basically the vendor name. Do you suggest some better term instead of `unique self-identifier`

Reply:
Just wanted to know what happens if that identifier happens to be not unique

Reply:
Nothing basically. it is for the user to uniquely identify a payment request if there are multiple requests from the same vendor. Whatever the value is, the wallet will parse and show it to the user. If it is not unique then the user might have some touble understanding and he will have to check the address and amount to find out what he is paying for.

Reply:
I think it might help to remove the `unique` then? Seems to be more restrictive than it actually ends up being. Maybe make it `human readable` identifier?

Reply:
sure. will do that.